circuit_knitting_toolbox/circuit_cutting/cutting_evaluation.py

```diff
@@ -82,3 +185,184 @@ def append_measurement_circuit(
         qc.measure(actual_qubit, obs_creg[clbit])
 
     return qc
+
+
+def _generate_cutting_experiments(
+    circuits: QuantumCircuit | dict[str | int, QuantumCircuit],
+    observables: PauliList | dict[str | int, PauliList],
+    num_samples: int,
+) -> tuple[list[list[list[QuantumCircuit]]], list[float], list[int]]:
+    """Generate all the experiments to run on the backend and their associated coefficients."""
+    # Retrieving the unique bases, QPD gates, and decomposed observables is slightly different
+    # depending on whether the decomposed circuit was separated into subcircuits before calling
+    # execute_experiments, but the 2nd half of this function can be shared between both cases.
+    if isinstance(circuits, QuantumCircuit):
+        is_separated = False
+        subcircuit_list = [circuits]
+        _, subsystem_observables = decompose_observables(
+            observables, "A" * len(observables[0])
+        )
+        # Gather the unique bases from the circuit
+        bases, qpd_gate_ids = _get_bases(circuits)
+        subcirc_qpd_gate_ids = [qpd_gate_ids]
+
+    else:
+        is_separated = True
+        subcircuit_list = [circuits[key] for key in sorted(circuits.keys())]
+        # Gather the unique bases across the subcircuits
+        subcirc_qpd_gate_ids, subcirc_map_ids = _get_mapping_ids_by_partition(
+            subcircuit_list
+        )
+        bases = _get_bases_by_partition(subcircuit_list, subcirc_qpd_gate_ids)
+
+        # Create the commuting observable groups
+        subsystem_observables = {
+            label: ObservableCollection(so) for label, so in observables.items()
+        }
+
+    # Sample the joint quasiprobability decomposition
+    random_samples = generate_qpd_samples(bases, num_samples=num_samples)
+
+    # Calculate terms in coefficient calculation
+    kappa = np.prod([basis.kappa for basis in bases])
+    num_samples = sum([value[0] for value in random_samples.values()])
+
+    # Sort samples in descending order of frequency
+    sorted_samples = sorted(random_samples.items(), key=lambda x: x[1][0], reverse=True)
+
+    # Generate the outputs -- sub-experiments, coefficients, and frequencies
+    subexperiments: list[list[list[QuantumCircuit]]] = []
+    coefficients = []
+    sampled_frequencies = []
+    for z, (map_ids, sample_weight) in enumerate(sorted_samples):
+        redundancy = sample_weight[0]
+        subexperiments.append([])
+        actual_coeff = np.prod(
+            [basis.coeffs[map_id] for basis, map_id in strict_zip(bases, map_ids)]
+        )
+        sampled_coeff = (redundancy / num_samples) * (kappa * np.sign(actual_coeff))
+        coefficients.append(sampled_coeff)
```

Comment:
Some day, we're also going to want the weight type of each coefficient later in the workflow.  I'm thinking something like

```suggestion
    for z, (map_ids, (redundancy, weight_type)) in enumerate(sorted_samples):
        subexperiments.append([])
        actual_coeff = np.prod(
            [basis.coeffs[map_id] for basis, map_id in strict_zip(bases, map_ids)]
        )
        sampled_coeff = (redundancy / num_samples) * (kappa * np.sign(actual_coeff))
        coefficients.append((sampled_coeff, weight_type))
```

Of course, this will need to be coupled with a change when the `coefficients` are processed later in the workflow.

If we do this now, we won't have to break this interface later.  But it's not strictly necessary to do this until we want to be able to propagate error bars.

(It's too bad the diff is not more compact; I edited the first and last lines, while removing the second.)